workspaces/marketplace/plugins/marketplace-common/src/types.ts

```diff
@@ -88,7 +108,13 @@ export interface MarketplacePluginSpec extends JsonObject {
  * @public
  */
 export interface MarketplaceApi {
-  getPlugins(): Promise<MarketplacePlugin[]>;
+  getPlugins(params: {
+    limit?: string;
+    cursor?: string;
+    sortByField?: string;
+    sortOrder?: 'asc' | 'desc';
+    searchText?: string;
+  }): Promise<MarketplacePluginWithPageInfo>;
```

Comment:
Please rename `params` to query and export the Params type as something like `GetPluginsRequest`.

Can you also reuse the types `EntityFilterQuery` and `EntityOrderQuery` for the filter and query. That allow us to just pass these values to the catalog client.

But this will require that you parse more complex types in the router. See comment there.

Can you please also use this type from https://backstage.io/docs/reference/catalog-client.queryentitiesinitialrequest/ for the full text search:

```tsx
    fullTextFilter?: {
        term: string;
        fields?: string[];
    };
```

The `request` param should be optional.